source/reactive-state.md

```diff
@@ -0,0 +1,124 @@
+```
+At any time you can get the state’s current value using `getReactiveStateValue('ExampleModalIsOpen')` which would in this case return `false`.
+
+You can set the state to a new value using `setReactiveStateValue('ExampleModalIsOpen', true)`.
+
+You can use a global state (like the one we created above) to make components reactive using [Apollo Client's useReactiveState hook](https://www.apollographql.com/docs/react/local-state/managing-state-with-field-policies/#storing-local-state-in-reactive-variables):
+```jsx
+const ExampelModal = ({title, children}) => {
+  const { ExampleModalIsOpen, setExampleModalIsOpen } = 
+      useReactiveState({ stateKey: 'ExampleModalIsOpen' });
+  
+  const handleOpen = () => {
+    setExampleModalIsOpen(true);
+  };
+  
+  const handleClose = () => {
+    setExampleModalIsOpen(false);
+  };
+  
+  return (
+    <div>
+      <Dialog onClose={handleClose} open={ExampleModalIsOpen}>
+          
+        <DialogTitle>{title}</DialogTitle>
+        
+        <DialogContent>{children}</DialogContent>
+        
+        <DialogActions>
+          <Button onClick={handleClose}>Close</Button>
+        </DialogActions>
+      
+      </Dialog>
+      
+      <Button onClick={handleOpen}>Open</Button>
+    </div>
+  );
+};
+```
+Now you may be thinking, _why can't I just use component state or `useState`?_ Well, you can, but if the component is ever unmounted and remounted, it will have lost its state. So, using the above example, if you open the modal, navigate to another route and then come back, the modal will no longer be open.
+
+Where you store your state has a lot to do with how persistent you need it to be:
+
+ * If you want that modal to start closed every time its mounted, store it in component state
+ * If you want the modal's open state to persist for the whole browser session, store it in reactive state
+ * If you want the state to persist even when the user comes back days later, store it in a field of the User document
+
+
+## Reactive state with an object and optional schema
+
+In addition to simple scalars, you can also store objects in reactive state:
+
+```js
+import { createReactiveState } from 'meteor/vulcan:core';
+
+const topBarSchema = {
+  messageId: {
+    type: String,
+    defaultValue: 'global.welcome',
+  },
+  showClose: {
+    type: Boolean,
+    defaultValue: true,
+  },
+  isOpen: {
+    type: Boolean,
+    defaultValue: false,
+  },
+  classes: {
+    type: Array,
+    optional: true,
+    arrayItem: {
+      type: String,
+    },
+  },
+};
+
+const topBarState = createReactiveState({ stateKey: 'TopBarState', schema: topBarSchema });
+```
+You probably don't need to assign the returned state object to a variable, but let's have a closer look at it. It has the following properties:
+
+ * `stateKey` - The name/id/key of the state
+ * `schema` - A `SimpleSchema` instance (or `undefined` if a schema was not passed when creating the state)
```

Comment:
From the document we probably want to know what it will be used for. I guess it will trigger an error if you create/mutate a variable and don't respect the schema? I would probably call that `simpleSchema` to clarify (in Vulcan NPM thanks to TS I've clarified the difference between a Vulcan Schema, a Simple Schema definition, and a SimpleSchema instance (= ExtendedSchema in their terminology).